cogeo_mosaic/backends/dynamodb.py

```diff
@@ -53,46 +80,56 @@ def assets_for_point(self, lng: float, lat: float) -> List[str]:
         tile = mercantile.tile(lng, lat, self.quadkey_zoom)
         return self.get_assets(tile.x, tile.y, tile.z)
 
-    def info(self, fetch_quadkeys: bool = False):
+    def info(self, quadkeys: bool = False):
         """Mosaic info."""
-        if not fetch_quadkeys:
-            warnings.warn(
-                "Returning empty quadkey list, performing full scan operation might be slow and expensive on large database."
-                "You can retrieve the list of quadkey by setting `fetch_quadkeys=True`"
-            )
-
         return {
             "bounds": self.mosaic_def.bounds,
             "center": self.mosaic_def.center,
             "maxzoom": self.mosaic_def.maxzoom,
             "minzoom": self.mosaic_def.minzoom,
             "name": self.mosaic_def.name if self.mosaic_def.name else "mosaic",
-            "quadkeys": [] if not fetch_quadkeys else self._quadkeys,
+            "quadkeys": [] if not quadkeys else self._quadkeys,
         }
 
     @property
     def _quadkeys(self) -> List[str]:
         """Return the list of quadkey tiles."""
-        warnings.warn(
-            "Performing full scan operation might be slow and expensive on large database."
+        resp = self.table.query(
+            KeyConditionExpression=Key("mosaicId").eq(self.mosaic_name),
+            ProjectionExpression="quadkey",
         )
-        resp = self.table.scan(ProjectionExpression="quadkey")  # TODO: Add pagination
-        return [qk["quadkey"] for qk in resp["Items"] if qk["quadkey"] != "-1"]
+        return [
+            qk["quadkey"]
+            for qk in resp["Items"]
+            if qk["quadkey"] != self._metadata_quadkey
+        ]
 
     def write(self, overwrite: bool = False, **kwargs: Any):
         """Write mosaicjson document to AWS DynamoDB."""
-        self._create_table(overwrite=overwrite, **kwargs)
+        if not self._table_exists():
+            self._create_table(**kwargs)
+
+        if self._mosaic_exists():
+            if not overwrite:
+                raise MosaicExists(
+                    f"Mosaic already exists in {self.table_name}, use `overwrite=True`."
+                )
+            self.clean()
+
         items = self._create_items()
         self._write_items(items)
 
     def _update_quadkey(self, quadkey: str, dataset: List[str]):
-        """Update quadkey list."""
-        self.table.put_item(Item={"quadkey": quadkey, "assets": dataset})
+        """Update single quadkey in DynamoDB."""
+        self.table.put_item(
+            Item={"mosaicId": self.mosaic_name, "quadkey": quadkey, "assets": dataset}
+        )
 
     def _update_metadata(self):
         """Update bounds and center."""
         meta = json.loads(json.dumps(self.metadata), parse_float=Decimal)
```

Comment:
Probably good to add a comment like "Note: this is required because DynamoDB requires all numbers to be in Decimal type"